metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/asa/BasicAsaParser.java

```diff
@@ -0,0 +1,209 @@
+    protected ZoneId deviceTimeZone;
+
+    private static final Map<String, String> patternMap = ImmutableMap.<String, String>builder()
+            .put("ASA-2-106001", "CISCOFW106001")
+		    .put("ASA-2-106006", "CISCOFW106006_106007_106010")
+		    .put("ASA-2-106007", "CISCOFW106006_106007_106010")
+		    .put("ASA-2-106010", "CISCOFW106006_106007_106010")
+		    .put("ASA-3-106014", "CISCOFW106014")
+		    .put("ASA-6-106015", "CISCOFW106015")
+		    .put("ASA-1-106021", "CISCOFW106021")
+		    .put("ASA-4-106023", "CISCOFW106023")
+		    .put("ASA-5-106100", "CISCOFW106100")
+		    .put("ASA-6-110002", "CISCOFW110002")
+		    .put("ASA-6-302010", "CISCOFW302010")
+		    .put("ASA-6-302013", "CISCOFW302013_302014_302015_302016")
+		    .put("ASA-6-302014", "CISCOFW302013_302014_302015_302016")
+		    .put("ASA-6-302015", "CISCOFW302013_302014_302015_302016")
+		    .put("ASA-6-302016", "CISCOFW302013_302014_302015_302016")
+		    .put("ASA-6-302020", "CISCOFW302020_302021")
+		    .put("ASA-6-302021", "CISCOFW302020_302021")
+		    .put("ASA-6-305011", "CISCOFW305011")
+		    .put("ASA-3-313001", "CISCOFW313001_313004_313008")
+		    .put("ASA-3-313004", "CISCOFW313001_313004_313008")
+		    .put("ASA-3-313008", "CISCOFW313001_313004_313008")
+		    .put("ASA-4-313005", "CISCOFW313005")
+		    .put("ASA-4-402117", "CISCOFW402117")
+		    .put("ASA-4-402119", "CISCOFW402119")
+		    .put("ASA-4-419001", "CISCOFW419001")
+		    .put("ASA-4-419002", "CISCOFW419002")
+		    .put("ASA-4-500004", "CISCOFW500004")
+		    .put("ASA-6-602303", "CISCOFW602303_602304")
+		    .put("ASA-6-602304", "CISCOFW602303_602304")
+		    .put("ASA-7-710001", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710002", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710003", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710005", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-7-710006", "CISCOFW710001_710002_710003_710005_710006")
+		    .put("ASA-6-713172", "CISCOFW713172")
+		    .put("ASA-4-733100", "CISCOFW733100")
+		    .put("ASA-6-305012", "CISCOFW305012")
+		    .put("ASA-7-609001", "CISCOFW609001")
+		    .put("ASA-7-609002", "CISCOFW609002")
+            .put("ASA-5-713041", "CISCOFW713041")
+            .build();
+
+    @Override
+    public void configure(Map<String, Object> parserConfig) {
+        String timeZone = (String) parserConfig.get("deviceTimeZone");
+        if (timeZone != null)
+            deviceTimeZone = ZoneId.of(timeZone);
+        else {
+            deviceTimeZone = ZoneOffset.UTC;
+            LOG.warn("[Metron] No device time zone provided; defaulting to UTC");
+        }
+    }
+
+    @Override
+    public void init() {
+        asaGrok = new Grok();
+        InputStream patternStream = this.getClass().getClassLoader().getResourceAsStream("patterns/asa");
+        try {
+            asaGrok.addPatternFromReader(new InputStreamReader(patternStream));
```

Comment:
Just calling this out for other's awareness, but this will only load the Grok pattern from the classpath.  The GrokParser seems to work much harder looking for a pattern.

Maybe that is what we want in this case, since changing the Grok pattern might not play well with the rest of the Java code involved with Kyle's parser.